system/system_monitor/src/gpu_monitor/gpu_monitor_base.cpp

```diff
@@ -50,38 +43,34 @@ GPUMonitorBase::GPUMonitorBase(const ros::NodeHandle & nh, const ros::NodeHandle
   updater_.add("GPU Frequency", this, &GPUMonitorBase::checkFrequency);
 }
 
-void GPUMonitorBase::run(void)
+void GPUMonitorBase::update()
 {
-  ros::Rate rate(1.0);
-
-  while (ros::ok()) {
-    ros::spinOnce();
     updater_.force_update();
-    rate.sleep();
-  }
 }
 
+void GPUMonitorBase::shut_down(){/*NOOP by default.*/}
+
 void GPUMonitorBase::checkTemp(diagnostic_updater::DiagnosticStatusWrapper & stat)
 {
-  ROS_INFO("GPUMonitorBase::checkTemp not implemented.");
+  RCLCPP_INFO(get_logger(), "GPUMonitorBase::checkTemp not implemented.");
```

Comment:
[MINOR]: (and likely not related) but I noticed that these logging messages are reported so frequently - should these be changes to a different logging style that is spams the console less such as: [RCLCPP_INFO_ONCE](http://docs.ros2.org/foxy/api/rclcpp/logging_8hpp.html#ac398a2db64fa466508754a6c4dc617c9)

Reply:
Good idea! Applied the suggestion.